src/test/java/io/kokuwa/maven/helm/UploadMojoTest.java

```diff
@@ -315,4 +369,78 @@ private void assertUpload(UploadMojo mojo, RequestMethod method, String path, St
+				.willReturn(WireMock.ok()
+						.withStatus(200)));
+		mock.stubFor(WireMock.get(WireMock.urlMatching(".*/index\\.yaml$"))
+				.willReturn(WireMock.ok()
+						.withStatus(200)
+						.withHeader("Content-Type", "application/yaml")
+						.withBody(getIndexYamlBody())));
+		mock.stubFor(WireMock.get(WireMock.urlMatching(".*\\.tgz$"))
+				.willReturn(WireMock.ok()
+						.withStatus(200)
+						.withHeader("Content-Type", "application/gzip")
+						.withBodyFile("app-0.1.0.tgz")));
+
+		assertDoesNotThrow(() -> mojo.execute(), "upload failed");
+		List<LoggedRequest> requests = mock.findAll(requestPattern);
+		assertEquals(3, requests.size(), "expected only three requests");
+		LoggedRequest request = requests.get(0);
+		assertEquals(method, request.getMethod(), "method");
+		assertEquals((mojo.getUploadRepoStable().getUrl().startsWith("https")
+				? "https://127.0.0.1:" + mock.getHttpsPort()
+				: "http://127.0.0.1:" + mock.getPort()) + path, request.getAbsoluteUrl(), "url");
+		assertEquals("application/gzip", request.getHeader(HttpHeaders.CONTENT_TYPE), "content-type");
+	}
+
+	private void assertUploadVerifyFail(UploadMojo mojo, RequestMethod method, String path) {
+		RequestPatternBuilder requestPattern;
+		requestPattern = RequestPatternBuilder.allRequests();
+
+		mock.stubFor(WireMock.put(WireMock.urlMatching(".*"))
+				.willReturn(WireMock.ok()
+						.withStatus(200)));
+		mock.stubFor(WireMock.get(WireMock.urlMatching(".*/index\\.yaml$"))
+				.willReturn(WireMock.ok()
+						.withStatus(200)
+						.withHeader("Content-Type", "application/yaml")
+						.withBody(getIndexYamlBody())));
+		mock.stubFor(WireMock.get(WireMock.urlMatching(".*\\.tgz$"))
+				.willReturn(WireMock.ok()
+						.withStatus(200)
+						.withHeader("Content-Type", "application/gzip")
+						.withBodyFile("app-0.1.0.tgz")));
+
+		assertThrows(MojoExecutionException.class, mojo::execute, "could not verify");
+		List<LoggedRequest> requests = mock.findAll(requestPattern);
+		LoggedRequest request = requests.get(0);
+		assertEquals(method, request.getMethod(), "method");
+		assertEquals((mojo.getUploadRepoStable().getUrl().startsWith("https")
+				? "https://127.0.0.1:" + mock.getHttpsPort()
+				: "http://127.0.0.1:" + mock.getPort()) + path, request.getAbsoluteUrl(), "url");
+		assertEquals("application/gzip", request.getHeader(HttpHeaders.CONTENT_TYPE), "content-type");
+	}
+
+	private String getIndexYamlBody() {
+		String indexYamlBody = "apiVersion: v1\n" +
+				"entries:\n" +
+				"  app:\n" +
+				"    - created: 2023-11-05T12:15:23.451853285-06:00\n" +
+				"      description: Dummy chart for testing\n" +
+				"      digest: digesthash1\n" +
+				"      home: https://helm.sh/helm\n" +
+				"      name: app\n" +
+				"      sources:\n" +
+				"        - https://github.com/helm/helm\n" +
+				"      urls:\n" +
+				"        - " + "http://127.0.0.1:" + mock.getPort() + "/nexus/app-0.1.0.tgz" + "\n" +
+				"      version: 0.1.0";
+		return indexYamlBody;
```

Comment:
we had to encode the mock index.yaml that is returned by the server inside this UploadMojoTest.java because a valid URL is required in the index.yaml and a dynamic port is set when testing. Thus, every time the test is run, the index.yaml must be consistent with whatever dynamic mock.getPort() returns